internal/engine/wazevo/engine.go

```diff
@@ -220,6 +220,7 @@ func (e *engine) compileModule(ctx context.Context, module *wasm.Module, listene
 	totalSize := 0 // Total binary size of the executable.
 	cm.functionOffsets = make([]int, localFns)
 	bodies := make([][]byte, localFns)
+	sourceOffsets := make([][]backend.SourceOffsetInfo, localFns)
```

Comment:
we need to append the sourceOffsets in the second stage, so we need to store `be.SourceOffsetInfo()` somewhere. I don't love this change, but I could not think of anything better at the moment...